nuget/lib/dependabot/nuget/file_parser/packages_config_parser.rb

```diff
@@ -26,16 +26,10 @@ def initialize(packages_config:)
         end
 
         def dependency_set
-          return parse_dependencies if CacheManager.caching_disabled?
-
           key = "#{packages_config.name.downcase}::#{packages_config.content.hash}"
           cache = PackagesConfigParser.dependency_set_cache
 
           cache[key] ||= parse_dependencies
-
-          dependency_set = Dependabot::FileParsers::Base::DependencySet.new
-          dependency_set += cache[key]
-          dependency_set
```

Comment:
Are we removing the caching behavior entirely with this change? I'm not quite following how removing this code doesn't alter behavior 🤔

Reply:
I should've extracted this to a separate commit. My understanding is that this is just redundant code. The result of the (cached or not) `parse_dependencies` method should be already a `Dependabot::FileParsers::Base::DependencySet`, unless I'm missing something.

Reply:
I believe the caching is done in the above line, which I kept!

Reply:
Thanks for clarifying!

Reply:
Awesome, I did split that commit out so the diff of the actual fix for "caching disabled mode" is easier to understand.

Also added one more missing require that running specs locally found.